lib/src/models/phone_number_range.dart

```diff
@@ -0,0 +1,76 @@
+import '../../phone_numbers_parser.dart';
+
+/// Describes a contiguous range of phone numbers
+class PhoneNumberRange {
+  final PhoneNumber start;
+  final PhoneNumber end;
+
+  const PhoneNumberRange(
+    this.start,
+    this.end,
+  );
+
+  @override
+  String toString() => '${start.international} - ${end.international}';
+
+  @override
+  bool operator ==(Object other) {
+    if (identical(this, other)) return true;
+
+    return other is PhoneNumberRange &&
+        other.start == start &&
+        other.end == end;
+  }
+
+  @override
+  int get hashCode => start.hashCode ^ end.hashCode;
+
+  /// Calculates the number of [PhoneNumber]s in the range (inclusive of start and end)
+  /// where 'this' is the start of the range and [endOfRange] is the end of the range.
+  int get count {
+    var first = BigInt.parse(_number(start));
+    var last = BigInt.parse(_number(end));
+
+    var bigInterval = last - first;
+
+    var interval = bigInterval.abs().toInt() + 1;
+
+    return interval;
+  }
+
+  String _number(PhoneNumber phone) => '${phone.dialCode}${phone.nsn}';
+
+  /// Returns a list of [PhoneNumber]s in the range
+  /// [this] to [endOfRange] inclusive.
+  List<PhoneNumber> expandRange() {
+    var startZeros = _getLeadingZeros(start);
+
+    var first = BigInt.parse(_number(start));
+    var last = BigInt.parse(_number(end));
+
```

Comment:
I'm having some trouble wrapping my head around some things here as it seems you have a future vision.

  - You used `_number` while there is an `international` property already. Since you added the leading zero I'm wondering if that's intentional.
  - Your comment above the leading zero check is about national number but here you are using the international version, which (currently) would only make sens for checking for international prefix starting with 0, (eg 00).
  
Unless I'm mistaking, something seems off ? Maybe the idea is to expand ranges of phone numbers without their dial code and with their national prefix, but that's not what's happening here.

Reply:
Going forward I would like to see support for national and local phone formats. The international format isn't particularly friendly for  end users. National and local formats are much easier to read.

In anticipation of support for local and national codes I want to support leading zeros.

The international method includes a '+' symbol which is why I strip it off and relates to leading zeros.

I guess the questions here is would we ever support a local number without knowing the dial code.
If we won't then there is no point supporting leading zeros.

Reply:
> I would like to see support for national and local phone formats.
  
  That's certainly possible. In another of your comment you mentioned formatting to its local format. I assumed a property like `.local` on the phone number class is what you had in mind.
  
However, your method expand range would return `PhoneNumber`s right ? You can then use the `.local` property on those. The leading zero check would still be done  on the `_number` which is starting with a dial code.
 
So, as far as I understand, it does not makes sens to use the leading zero check here as it is only confusing as it is currently misapplied.

 > the questions here is would we ever support a local number without knowing the dial code.
 
 By " without knowing the dial code " I assume that you mean that the library does not try to figure out where a phone number originates from (in comparison, `parseRaw` does). If that's the case, the library is not aware of the phone metadata, which means you cannot transform, validate or learn anything about the phone number. It is just a string. The only thing that could be used is the range but at that point it sounds more like string manipulation than phone number parsing.
 
 > The international method includes a '+' symbol which is why I strip it off and relates to leading zeros.

Does it matter ? `BigInt.parse('+45') == BigInt.parse('45')`